src/registry/textarea/edit.tsx

```diff
@@ -0,0 +1,243 @@
+import {useFormikContext} from 'formik';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {
+  AutoComplete,
+  BuilderTabs,
+  ClearOnHide,
+  Description,
+  Hidden,
+  IsSensitiveData,
+  Key,
+  Label,
+  Multiple,
+  Placeholder,
+  PresentationConfig,
+  ReadOnly,
+  ShowCharCount,
+  SimpleConditional,
+  Tooltip,
+  Translations,
+  Validate,
+  useDeriveComponentKey,
+} from '@/components/builder';
+import {LABELS} from '@/components/builder/messages';
+import {Checkbox, NumberField, TabList, TabPanel, Tabs, TextArea} from '@/components/formio';
+import {getErrorNames} from '@/utils/errors';
+
+import {EditFormDefinition} from '../types';
+
+/**
+ * Form to configure a Formio 'textarea' type component.
+ */
+const EditForm: EditFormDefinition<TextareaComponentSchema> = () => {
+  const intl = useIntl();
+  const [isKeyManuallySetRef, generatedKey] = useDeriveComponentKey();
+  const {values, errors} = useFormikContext<TextareaComponentSchema>();
+
+  const erroredFields = Object.keys(errors).length
+    ? getErrorNames<TextareaComponentSchema>(errors)
+    : [];
+  // TODO: pattern match instead of just string inclusion?
+  // TODO: move into more generically usuable utility when we implement other component
+  // types
+  const hasAnyError = (...fieldNames: string[]): boolean => {
+    if (!erroredFields.length) return false;
+    return fieldNames.some(name => erroredFields.includes(name));
+  };
+
+  Validate.useManageValidatorsTranslations<TextareaComponentSchema>([
+    'required',
+    'maxLength',
+    'pattern',
+  ]);
+  return (
+    <Tabs>
+      <TabList>
+        <BuilderTabs.Basic
+          hasErrors={hasAnyError(
+            'label',
+            'key',
+            'description',
+            'tooltip',
+            'showInSummary',
+            'showInEmail',
+            'showInPDF',
+            'multiple',
+            'hidden',
+            'clearOnHide',
+            'isSensitiveData',
+            'defaultValue',
+            'autocomplete',
+            'disabled',
+            'placeholder',
+            'showCharCount',
+            'autoExpand',
+            'rows'
+          )}
+        />
+        <BuilderTabs.Advanced hasErrors={hasAnyError('conditional')} />
+        <BuilderTabs.Validation hasErrors={hasAnyError('validate')} />
+        <BuilderTabs.Translations hasErrors={hasAnyError('openForms.translations')} />
+      </TabList>
+
+      {/* Basic tab */}
+      <TabPanel>
+        <Label />
+        <Key isManuallySetRef={isKeyManuallySetRef} generatedValue={generatedKey} />
+        <Description />
+        <Tooltip />
+        <PresentationConfig />
+        <Multiple<TextareaComponentSchema> />
+        <Hidden />
+        <ClearOnHide />
+        <IsSensitiveData />
+        <DefaultValue multiple={!!values.multiple} />
+        <AutoComplete />
+        <ReadOnly />
+        <Placeholder />
+        <ShowCharCount />
+        <AutoExpand />
```

Comment:
I don't know if `AutoExpand` could be reflected in the preview? It doesn't seem to be supported in plain HTML

Reply:
Looking at https://stackoverflow.com/a/65990608 it doesn't seem to be a lot of work, could easily be implemented as a generic hook even.

W/r to this builder it's a bit of a tradeoff - if it's not a lot of work, yep, let's get the behaviour implemented (the added benefit is that it informs us for the future implementation in the renderer), if it's too much work then we might get by with making it appear similarly (e.g. the `inputMask` functionality is mimicked via `placeholder` attribute, since it's waaaay too much work and has accessibility issues too).

Reply:
The SO answer worked fine